### ISSUE TITLE
Issues starting new clusters with backups

### ISSUE THREAD

**Issue by mjpitz** [bitpoke/mysql-operator#623]
My configuration seems pretty straight forward but my cluster isn't starting for what looks like permissions issues

```
020-10-28T17:46:04.777011213Z  ERROR   sidecar clone command failed    {"error": "removing lost+found: openfdat /var/lib/mysql/lost+found: permission denied"}
```

I tried enabling the volume permission container but that doesn't do anything since it's last in the chain and never gets a chance to run. Can someone help me understand why my cluster won't start?

```
---
apiVersion: mysql.presslabs.org/v1alpha1
kind: MysqlCluster
metadata:
  namespace: xxx
  name: xxx
spec:
  replicas: 2
  secretName: xxx

  backupSchedule: "0 0 * * *"
  backupURL: s3://backups/xxx/
  backupSecretName: xxx-backup
  backupRemoteDeletePolicy: retain

  podSpec:
    resources:
      requests:
        cpu: 500m
        memory: 1Gi
      limits:
        cpu: 500m
        memory: 1Gi
    initContainers:
      - name: volume-permissions
        image: busybox
        securityContext:
          runAsUser: 0
        command:
          - sh
          - -c
          - chmod 750 /data/mysql; chown 999:999 /data/mysql
        volumeMounts:
          - name: data
            mountPath: /data/mysql
  volumeSpec:
    persistentVolumeClaim:
      accessModes: [ "ReadWriteOnce" ]
      resources:
        requests:
          storage: 5Gi
```

**Comment by mjpitz** [comment]
So I've disabled backups and am still having this problem with a new cluster.

**Comment by mjpitz** [comment]
Stack trace from init container causing crashloop backoff: 

```
github.com/presslabs/mysql-operator/vendor/github.com/go-logr/zapr.(*zapLogger).Error
        /go/src/github.com/presslabs/mysql-operator/vendor/github.com/go-logr/zapr/zapr.go:128
main.main.func2
        /go/src/github.com/presslabs/mysql-operator/cmd/mysql-operator-sidecar/main.go:67
github.com/presslabs/mysql-operator/vendor/github.com/spf13/cobra.(*Command).execute
        /go/src/github.com/presslabs/mysql-operator/vendor/github.com/spf13/cobra/command.go:766
github.com/presslabs/mysql-operator/vendor/github.com/spf13/cobra.(*Command).ExecuteC
        /go/src/github.com/presslabs/mysql-operator/vendor/github.com/spf13/cobra/command.go:852
github.com/presslabs/mysql-operator/vendor/github.com/spf13/cobra.(*Command).Execute
        /go/src/github.com/presslabs/mysql-operator/vendor/github.com/spf13/cobra/command.go:800
main.main
        /go/src/github.com/presslabs/mysql-operator/cmd/mysql-operator-sidecar/main.go:111
runtime.main
        /usr/local/go/src/runtime/proc.go:203
```

It's really frustrating because it seems like it up and stopped working on me.

**Comment by AMecea** [comment]
Hi @mjpitz, I think the issues is that the init containers that you have specified are run after the mysql-operator containers. This is fixed and will be released in `0.5` (see #600).

**Comment by mjpitz** [comment]
@AMecea thanks for following up! I think there might also be something in play from my cloud providers perspective too. Haven't been able to prove anything yet, but ran into issues provisioning a redis cluster earlier.

I've been able to mitigate in the meantime.
